forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -6975,6 +6983,7 @@ public void onEndOfCombat(final Player active) {
             chosenModesYourCombat.clear();
             chosenModesYourCombatStatic.clear();
             updateAbilityTextForView();
+            enlistedThisCombat = false;
```

Comment:
imo you need to move this out of the branch to make sure it gets always reset in face of controller change